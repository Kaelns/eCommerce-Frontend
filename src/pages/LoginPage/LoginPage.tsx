import { Stack } from '@mui/system';

import { TitleTypography } from '@/shared/ui/elements/typography/TitleTypography';

export function LoginPage() {
  return (
    <Stack justifyContent="center" alignItems="center" gap={2}>
      <TitleTypography>Login page</TitleTypography>
      {/* <LoginForm /> */}
    </Stack>
  );
}
