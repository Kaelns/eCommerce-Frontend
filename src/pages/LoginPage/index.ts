export { LoginPage } from '@/pages/LoginPage/LoginPage';
