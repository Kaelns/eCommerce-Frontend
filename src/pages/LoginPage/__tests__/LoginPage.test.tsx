import { LoginPage } from '@/pages/LoginPage/LoginPage';
import { render } from '@testing-library/react';
import * as useAuthContextHook from '@/context/AuthContext/useAuthContext';

describe('Given LoginPage component: ', () => {
  it('when rendered, should match snapshot', () => {
    const useMoviesSpy = vi.spyOn(useAuthContextHook, 'useAuthContext');

    useMoviesSpy.mockReturnValue({
      authTokens: {
        token: '',
        anonToken: '',
        refreshToken: ''
      },
      setAuthTokens: vi.fn()
    });

    const { asFragment } = render(<LoginPage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
