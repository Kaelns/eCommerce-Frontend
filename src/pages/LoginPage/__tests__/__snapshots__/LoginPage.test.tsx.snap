// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Given LoginPage component:  > when rendered, should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root css-13pq5fk-MuiStack-root"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 css-1yej70y-MuiTypography-root"
    >
      Login page
    </h5>
    <form
      class="MuiStack-root css-15vlcj4-MuiStack-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-animated css-1d2t4i-MuiFormLabel-root-MuiInputLabel-root"
      >
        Enter your email
        <span
          aria-hidden="true"
          class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           *
        </span>
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary css-j882ge-MuiInputBase-root-MuiOutlinedInput-root"
        required=""
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input css-1jk99ih-MuiInputBase-input-MuiOutlinedInput-input"
          name="email"
          required=""
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-gpjuor"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        class="MuiFormHelperText-root Mui-error css-1d1r5q-MuiFormHelperText-root"
      />
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-animated css-1d2t4i-MuiFormLabel-root-MuiInputLabel-root"
      >
        Enter your password
        <span
          aria-hidden="true"
          class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           *
        </span>
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-adornedEnd css-lxfshk-MuiInputBase-root-MuiOutlinedInput-root"
        required=""
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-lc42l8-MuiInputBase-input-MuiOutlinedInput-input"
          name="password"
          required=""
          type="password"
          value=""
        />
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium MuiInputAdornment-root MuiInputAdornment-positionEnd css-1yrfzg2-MuiButtonBase-root-MuiIconButton-root-MuiInputAdornment-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-18w7uxr-MuiSvgIcon-root"
            data-testid="VisibilityIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5M12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5m0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3"
            />
          </svg>
        </button>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-gpjuor"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        class="MuiFormHelperText-root Mui-error css-1d1r5q-MuiFormHelperText-root"
      />
      <div
        class="MuiStack-root css-jz9jx9-MuiStack-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-12b4no2-MuiButtonBase-root-MuiButton-root"
          disabled=""
          tabindex="-1"
          type="submit"
        >
          Login
        </button>
        <div
          aria-orientation="horizontal"
          class="MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren css-1tscyog-MuiDivider-root"
          role="separator"
        >
          <span
            class="MuiDivider-wrapper css-qywfm8-MuiDivider-wrapper"
          >
            <div
              class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-14tabw5-MuiChip-root"
            >
              <span
                class="MuiChip-label MuiChip-labelSmall css-3fh3fk-MuiChip-label"
              >
                Or
              </span>
            </div>
          </span>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-142372x-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Register
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
